contributing/pre-commit

```diff
@@ -11,4 +11,7 @@ printf "running flake8. This script will preempt the commit if flake8 fails.\n"
 flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py
 printf "flake8 passed!\n"
 
+printf "running mypy. This script will preempt the commit if mypy fails.\n"
```

Comment:
I think I'm the only person who uses this hook :) We should add the MyPy check to [CI](https://github.com/modin-project/modin/blob/3982306a540bd09b43a21dcdca6b16b4741ed151/.github/workflows/ci.yml). Is there a way to have it check all changed lines in other files, in addition to all of `**/modin/logging/*.py`?

Reply:
But we should also keep this change! I know I will want to add this to my pre-commit hook.

Reply:
I use this hook too!

Reply:
Haha, I actually use this pre-commit hook. It's super useful! And yes, I intend on adding the MyPy check to CI. I'm not sure if we can have it check all changed lines in other files. I think it would be hard for mypy to do it only on recent diffs. I think we're going to have to slowly make type changes over time, but I'll try to see if I can find anything online about this.